### PR TITLE
feat(moderations): Paginated view for requests tab for moderations

### PR DESCRIPTION
**Pull request by smrutis1**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.

> * Which issue is this pull request belonging to and how is it solving it? (*Refer to issue here*)

Issue: 

It is observed the "Requests" tab taking longer time to load if the number of requests are higher.  This changes adds the paginated view response for loading moderation requests instead off getting all the moderations on a single hit upon clicking on "Requests" tab which might decrease the loading time. 

Note: 
- Had to remove the data table quick filter and add the advanced search as it is there in Components and Projects tab to make the search work properly. 
- Paginated response for clearing requests are not taken care of as part of it as per now.

> * Did you add or update any new dependencies that are required for your change?

Need to test the moderation functionality. 

> Have you implemented any additional tests? No

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

Signed-off-by: Smruti Prakash Sahoo <smruti.sahoo@siemens.com>

### CODE REVIEW

File: frontend/sw360-portlet/src/main/resources/META-INF/resources/html/moderation/view.jsp [hunk 1]
```diff
@@ -164,7 +239,18 @@
...
+                            <colgroup>
+                                <col />
+                                <col />
+                                <col style="width: 25%;" />
+                                <col style="width: 20%;" />
+                                <col style="width: 20%;" />
+                                <col style="width: 35%;" />
+                                <col />
+                                <col style="width: 1.7rem;" />
+                            </colgroup>
```
Comment (JaideepPalit) [inline frontend/sw360-portlet/src/main/resources/META-INF/resources/html/moderation/view.jsp:252]: Is it possible ` <colgroup>` can be removed and handle it while creating Datatable, like passing width parameter similar to line 502 ?
Indentation between Table and colgroup

**Review by JaideepPalit — CHANGES_REQUESTED** [review]
Reviewed the PR.
Kindly have a look at some minor review comments

File: frontend/sw360-portlet/src/main/resources/META-INF/resources/html/moderation/view.jsp (hunk 1 above)
Comment (smrutis1) [reply frontend/sw360-portlet/src/main/resources/META-INF/resources/html/moderation/view.jsp]: Done

**Comment by smrutis1** [comment]
@JaideepPalit , Thanks for review. Have incorporated the review comments.

**Comment by JaideepPalit** [comment]
Tested the PR in stage.
Seems to be some issue with pagination.
Issue 1 - 
  1. Search using Advanced filter like moderators. Results in maybe 100+ records.
  2. Click on second or third Page.
  3. List now contains more than 10 records even though max record per page is 10
  4. Shared Screenshot over mail
  
Issue 2 -
  - There is no drop down for Department in Advanced filter like in Project List Page
image

Issue 3 (Position of Search Dialog)-
  - Click Open/Closed CR. Search Dialog lies at the end after Page Menu. Same should be maintained in Open/Closed Mod Request
image

**Comment by smrutis1** [comment]
> Issue 1 -

Fixed the issues with no of records upon page navigation.

Note: Pagination is not applied on search results(will get all search results at once from DB) , hoping the result will contain limited number of data and should not effect the loading time that much.

> Issue 2 -

Removed text based search and added the dropdown with prefilled value of requesting users department. 

> Issue 3 (Position of Search Dialog)-

Moved the search box below the menu.

**Comment by JaideepPalit** [comment]
Retested the PR on stage.
- Issue 1 - fixed
- Issue 3 - fixed
- Issue 2 - persists.
   - Dropdown for Department is coming empty. Error Logs - [error.txt](https://github.com/eclipse/sw360/files/6559148/error.txt)
departmentDropDown

**Comment by JaideepPalit** [comment]
Few Points related to Dropdown for Department
- It would be good to have sorted list of dropdown alphabetically instead of randomly ordered list, could be achieved by `TreeSet`.
- Similar to Project List Page, It would be good to have list of departments for which moderation request exists, instead of all possible values.

**Comment by smrutis1** [comment]
> 
> * It would be good to have sorted list of dropdown alphabetically instead of randomly ordered list, could be achieved by `TreeSet`.

Fixed it. Thanks for suggestion.

> * Similar to Project List Page, It would be good to have list of departments for which moderation request exists, instead of all possible values.

(from discussion) : Not an issue as it fetching the user's department from moderation requests itself.

**Comment by mcjaeger** [comment]
tested: indeed it seems like the groups are only taken from the open moderation requests and with few groups it appears sorted

**Pull request merged**